www/perl-lib/SMT/NCCRegTools.pm

```diff
@@ -103,8 +103,30 @@ sub new
     $self->{URI}      = $ruri;
     $self->{AUTHUSER} = $user;
     $self->{AUTHPASS} = $pass;
-    bless($self);
 
+    my $regsharing = SMT::Utils::hasRegSharing();
+    if (! $regsharing) {
+        my $msg = 'Could not read SMT configuration file';
+        printLog($self->{LOG}, $self->vblevel(), LOG_WARN, __($msg));
+        $regsharing = 'false';
+    }
+    elsif ($regsharing eq 'true') {
+        eval
+        {
+            require 'SMT/RegistrationSharing.pm';
+        };
+        if ($@)
+        {
+            my $msg = 'Failed to load registration sharing module '
+              . '"SMT/RegistrationSharing.pm"'
+              . "\n$@";
+            printLog($self->{LOG}, $self->vblevel(), LOG_ERROR, __($msg));
+            $regsharing = 'false';
+        }
+    }
+    $self->{REGSHARING} = $regsharing;
```

Comment:
You need to re-think the assignment of this variable.
It has now 1 or the string 'false'.
Make it 0/1 everywhere and change the checks also at other places of this file.